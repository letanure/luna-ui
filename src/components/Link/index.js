import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

/**
 * Link component.
 * E.g.:
 * ```html
 * <Link href="http://example.de" target="_blank" />
 * ```
 */
class LinkComponent extends React.PureComponent {
  render() {
    const Link = styled.a`
      transition: opacity linear .25s;
      color: ${(props) => props.theme.linkColor};
      &:hover {
        opacity: .8;
      }
    `;

    return <Link {...this.props}>{this.props.children}</Link>;
  }
}

LinkComponent.propTypes = {
  children: PropTypes.node
};

LinkComponent.defaultProps = {
  children: null
};

export default LinkComponent;
