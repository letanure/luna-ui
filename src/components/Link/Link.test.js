import React from 'react';
import renderer from 'react-test-renderer';

import Link from './index';

describe('<Link />', () => {
  const component = <Link href="http://example.de">My Link</Link>;

  test('toMatchSnapshot', () => {
    const tree = renderer
      .create(component)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
