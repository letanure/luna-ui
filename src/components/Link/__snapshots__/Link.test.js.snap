// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> toMatchSnapshot 1`] = `
<a
  className="sc-bdVaJa eZStWY"
  href="http://example.de"
>
  My Link
</a>
`;
